src/amo/containers/App.js

```diff
@@ -43,10 +47,28 @@ export class AppBase extends React.Component {
     MastHeadComponent: MastHead,
     _addChangeListeners: addChangeListeners,
     mozAddonManager: config.get('server') ? {} : navigator.mozAddonManager,
+    userAgent: null,
   }
 
   componentDidMount() {
-    const { _addChangeListeners, handleGlobalEvent, mozAddonManager } = this.props;
+    const {
+      _addChangeListeners,
+      handleGlobalEvent,
+      mozAddonManager,
+      setUserAgent,
+      userAgent,
+    } = this.props;
+
+    // If userAgent isn't set in state it could be we couldn't get one from
+    // the request headers on our first (server) request. If that's the case
+    // we try to load them from navigator.
+    const _navigator = navigator || window.navigator;
```

Comment:
I did that and then removed it because I think elsewhere we didn't have a test for navigator? But yeah good point, I'll add that back and run a test.